Rework shell help (#11664)
zephyrproject-rtos/zephyr: doc/subsystems/shell/shell.rst

	}

.. warning::
	Do not use function :cpp:func:`shell_fprintf` or shell print macros:
	:c:macro:`shell_print`, :c:macro:`shell_info`, :c:macro:`shell_warn`,
	:c:macro:`shell_error` outside of the command handler because this
	might lead to incorrect text display on the screen. If any text should
	be displayed outside of the command context, then use the
	:ref:`logger` or `printk` function.

Command help
------------

Every user-defined command or subcommand can have its own help description.
The help for commands and subcommands can be created with respective macros:

> :c:macro:`SHELL_CMD_REGISTER`, :c:macro:`SHELL_CMD_ARG_REGISTER`,

Reviewer: It can be updated in separate PR.
If you will register a command with `SHELL_CMD_ARG_REGISTER` shell will check if command has be called with required number of arguments. You will not need to do it in command handler

> :c:macro:`SHELL_CMD_REGISTER`, :c:macro:`SHELL_CMD_ARG_REGISTER`,

Reviewer 2: ARG variant takes the number of mandatory and optional arguments so they can be validated with use of precheck.

> :c:macro:`SHELL_CMD_REGISTER`, :c:macro:`SHELL_CMD_ARG_REGISTER`,

Reviewer: I've updated macro description

> :c:macro:`SHELL_CMD` and :c:macro:`SHELL_CMD_ARG`.

Reviewer 3: ```suggestion
:c:macro:`SHELL_CMD`, and :c:macro:`SHELL_CMD_ARG`.
```

Reviewer: fixed

> Shell will print defined help message if user will call command

Reviewer 3: ```suggestion
Shell prints this help message when you call a command
```

Reviewer: fixed

		/* If it is a subcommand handler parent command syntax
		 * can be found using argv[-1].

> 		 */

Reviewer: fixed

> 		shell_print(shell, "This command has a parent command: %s",

Reviewer: @Vudentz : Do you have any opinion about that?

> 		shell_print(shell, "This command has a parent command: %s",

Reviewer 2: Don't have a strong opinion, either print or printf are ok with me, though if we change it back to printf we will need to change all the places using it.

> 		shell_print(shell, "This command has a parent command: %s",

Reviewer: I will keep it as it is. In my opinion it makes limited sense to change complementary macros:
like `shell_error` to `shell_errof` ;)
